src/c++/perf_analyzer/genai-perf/genai_perf/llm_inputs/dataset_retriever.py

```diff
@@ -57,25 +60,56 @@ def from_url(url: str, starting_index: int, length: int) -> List[Dict[str, Any]]
         return formatted_rows
 
     @staticmethod
-    def from_file(file_path: Path) -> List[Dict[str, str]]:
-        with open(file_path, "r") as file:
-            data = [load_json_str(line) for line in file]
+    def from_file(file_path: Path, output_format: OutputFormat) -> List[Dict[str, str]]:
+        contents = DatasetRetriever._load_file_content(file_path)
+
+        dataset = []
+        for content in contents:
+            data = {"text_input": content.get("text_input", "")}
+
+            if output_format == OutputFormat.OPENAI_VISION:
```

Comment:
If we're using the same image every time, we should do it outside of the for loop. Noting this here, though we might want to move this functionality elsewhere anywhere (mentioned in above comment).

Reply:
Unsure if I understood your comment. This part of code is reading the image files specified in the input file so the images could be different.

Reply:
Oh, I see. Thanks for explaining!